File: Brute Force/Insertion Sort/code.cpp
#include "algorithm-visualizer.h"
#include <iostream>

//visualizer{
const int				ARRAYSIZE = 15;
ChartTracer				chart("ChartTracer");
Array1DTracer			tracer("Array1DTracer");
LogTracer				logger("LogTracer");
Randomize::Array1D<int> array1D(ARRAYSIZE);
int						D[ARRAYSIZE];

void InitInsertionSort()
{
	Layout::setRoot(VerticalLayout({ chart,tracer,logger }));
	array1D.fill(D);
	tracer.set(D);
	tracer.chart(chart);
	Tracer::delay();
}
//}

int main()
{
	InitInsertionSort();
	//visualizer{
	logger.print("original array = [");
	for (int i = 0; i < ARRAYSIZE-1; i++)
	{
		logger.print(std::to_string(D[i]) + " , ");
	}
	logger.println(std::to_string(D[ARRAYSIZE-1]) + " ] ");
	//}

	for (int i = 1; i < ARRAYSIZE; i++) {
		int key = D[i];
		//visualizer{
		logger.println("insert" + std::to_string(key));
		tracer.select(i);
		Tracer::delay();
		//}
		int j;
		for (j = i - 1; (j >= 0) && (D[j] > key); j--) {
			D[j + 1] = D[j];
			//visualizer{
			tracer.patch(j + 1, D[j + 1]);
			Tracer::delay();
			tracer.depatch(j + 1);
			//}
		}
		D[j + 1] = key;
		//visualizer{
		tracer.patch(j + 1, D[j + 1]);
		Tracer::delay();
		tracer.depatch(j + 1);
		tracer.deselect(i);
		//}
	}
	//visualizer{
	logger.print("sorted array = [");
	for (int i = 0; i < ARRAYSIZE-1; i++)
	{
		logger.print(std::to_string(D[i]) + " , ");
	}
	logger.println(std::to_string(D[ARRAYSIZE-1]) + " ] ");
	//}
}

File: Brute Force/Insertion Sort/Code.java
import org.algorithm_visualizer.*;
import java.util.Arrays;

class Main {
    //visualizer{
    private static ChartTracer chartTracer = new ChartTracer();
    private static LogTracer logTracer = new LogTracer("Console");
    //}
    private static int sizeOfArray = 15;
    private static int minValue = 1;
    private static int maxValue = 20;
    private static Integer [] array = (Integer[]) new Randomize.Array1D(sizeOfArray, new Randomize.Integer(minValue, maxValue)).create();

    public static void main(String[] args) {
        int length = array.length;
        //visualizer{
        Layout.setRoot(new VerticalLayout(new Commander[]{chartTracer, logTracer}));
        logTracer.printf("original array = %s\n", Arrays.toString(array));
        chartTracer.set(array);
        Tracer.delay();
        //}
        for (int i = 1; i < length; i++) 
        {
            int j = i - 1;
            int temp = array[i];
            //visualizer{
            chartTracer.select(i);
            Tracer.delay();
            logTracer.printf("insert %s\n",temp);
            //}
            while (j >= 0 && array[j] > temp)
             {
                array[j + 1] = array[j];
                //visualizer{
                chartTracer.patch(j + 1, array[j + 1]);
                Tracer.delay();
                chartTracer.depatch(j + 1);
                //}
                j--;
            }
            array[j + 1] = temp;
            //visualizer{
            chartTracer.patch(j + 1, temp);
            Tracer.delay();
            chartTracer.depatch(j + 1);
            chartTracer.deselect(i);
            //}
        }
        logTracer.printf("sorted array = %s\n",Arrays.toString(array));
    }
}
